src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -45,91 +48,102 @@ public ErrorLogger(Stream stream, string toolName, string toolFileVersion, Versi
 
         private void WriteToolInfo(string name, string fileVersion, Version assemblyVersion)
         {
-            _writer.WriteObjectStart("toolInfo");
+            _writer.WriteObjectStart("tool");
             _writer.Write("name", name);
             _writer.Write("version", assemblyVersion.ToString(fieldCount: 3));
             _writer.Write("fileVersion", fileVersion);
             _writer.WriteObjectEnd();
         }
 
-        internal void LogDiagnostic(Diagnostic diagnostic, CultureInfo culture)
+        public void LogDiagnostic(Diagnostic diagnostic)
         {
             _writer.WriteObjectStart(); // result
             _writer.Write("ruleId", diagnostic.Id);
-            _writer.Write("kind", GetKind(diagnostic.Severity));
 
-            WriteLocations(diagnostic.Location, diagnostic.AdditionalLocations);
-
-            string message = diagnostic.GetMessage(culture);
-            if (string.IsNullOrEmpty(message))
+            string ruleIdKey = _descriptors.Add(diagnostic.Descriptor);
+            if (ruleIdKey != diagnostic.Id)
             {
-                message = "<None>";
+                _writer.Write("ruleIdKey", ruleIdKey);
             }
 
-            string description = diagnostic.Descriptor.Description.ToString(culture);
-            if (string.IsNullOrEmpty(description))
+            _writer.Write("level", GetLevel(diagnostic.Severity));
+
+            string message = diagnostic.GetMessage(_culture);
+            if (!string.IsNullOrEmpty(message))
             {
-                _writer.Write("fullMessage", message);
+                _writer.Write("message", message);
             }
-            else
+
+            if (diagnostic.IsSuppressed)
             {
-                _writer.Write("shortMessage", message);
-                _writer.Write("fullMessage", description);
+                _writer.WriteArrayStart("suppressionStates");
+                _writer.Write("suppressedInSource");
```

Comment:
It's not too late since the latest schema hasn't been published. Should I bump to beta.5 in next commit where I do this and adjust tags to be under properties?